Source/Core/ProverUtil.cs

```diff
@@ -301,7 +302,8 @@ public abstract class ProverFactory
   {
     // Really returns ProverInterface.
     //public abstract object! SpawnProver(ProverOptions! options, object! ctxt);
-    public abstract object SpawnProver(SMTLibOptions libOptions, ProverOptions options, object ctxt);
+    public abstract object SpawnProver(SMTLibOptions libOptions, ProverOptions options, object ctxt,
+      CancellationToken cancellationToken);
```

Comment:
Is it a problem to break this API?  Existing implementations of this interface will have to gain an extra argument.
(Alternatively, I don't know what `ctxt` is, but maybe it could contain the cancellationToken)